### PR TITLE
ci: fail fast on release branch

### PR DESCRIPTION
**Pull request by jaredperreault-okta**
## PR Checklist

- [ ] Have you verified the basic functionality for this change?
- [ ] Did you add tests, as appropriate, following our [Automated Test guidelines](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/2676497890/Automated+Testing+in+the+Signin+Widget)?
- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?
- [ ] Did you verify the change by running [downstream monolith artifact](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/102897979/Sign-in+Widget+Development#Sign-inWidgetDevelopment-Instructionstocreateandrunthedownstreamartifact(d16t))? (Provide link to build below)
- [ ] Does this PR include noticeable changes to the UI? (If yes, attach screenshots/video below)

### Issue:

- [OKTA-XXXXXX](https://oktainc.atlassian.net/browse/OKTA-XXXXXX)

### CODE REVIEW

File: test/e2e/runner.js [hunk 1]
```diff
@@ -93,6 +97,12 @@ function runNextTask() {
   const task = tasks.shift();
   console.log(`Running next task: ${task.description}`);
   task().then((code) => {
+    if (IS_RELEASE_BRANCH) {
```
Comment (denysoblohin-okta) [inline test/e2e/runner.js:100]:
```suggestion
    if (IS_RELEASE_BRANCH && code) {
```
?

File: test/e2e/runner.js [hunk 2]
```diff
@@ -93,6 +97,12 @@ function runNextTask() {
   const task = tasks.shift();
   console.log(`Running next task: ${task.description}`);
   task().then((code) => {
+    if (code !== 0 && IS_RELEASE_BRANCH) {
+      console.log('###\nRELEASE BRANCH DETECTED, EXITTING EARLY DUE TO FAILURE\n###');
```
Comment (shawyu-okta) [inline test/e2e/runner.js:101]:
```suggestion
      console.log('###\nRELEASE BRANCH DETECTED, EXITING EARLY DUE TO FAILURE\n###');
```

File: test/e2e/runner.js [hunk 3]
```diff
@@ -93,6 +97,12 @@ function runNextTask() {
...
+      // eslint-disable-next-line no-process-exit
+      process.exit(1);
```
Comment (shawyu-okta) [inline test/e2e/runner.js:103]: Should we exit with `code` instead of always `1`?

**Review by shawyu-okta — APPROVED** [review]
minor comments

**Review by lesterchoi-okta — APPROVED** [review]
Consider pulling the regex and other magic strings out to shared constants module?

**Pull request merged**